active-rfcs/0000-finalize-head-body.md

```diff
@@ -0,0 +1,80 @@
+  - Caused Astro to create different output for a component based on the file path of that component.
+
+## Goals
+
+1. Agree on and document the rules and behaviors of the `<head>`, `<body>`, `<html>`, and `<!DOCTYPE html>` tags.
+2. Remove special compiler behavior based on where a file lives in your src directory. 
+3. Remove `src/layouts` as a "special" folder and any config around this.
+
+## Non-Goals
+
+1. **Out of scope:** Support `<head>` Injection inside of individual components. This is a commonly requested user feature, but can be tackled as an additional feature, leaving this RFC focused on clarifying our existing API. This RFC does not impact that feature request.
+
+
+# Detailed design
+
+## Template Changes
+
+- `<!DOCTYPE html>`
+  - compiler: completely ignored by the compiler, stripped out from render output.
+  - runtime: a `<!DOCTYPE html>` tag is always included in final page HTML output.
+- `<html>`, `<body>`, `<head>`
+  - compiler: will be left as-is in the component template.
+  - runtime: will be left as-is in final page HTML output.
+  - runtime: may warn if duplicate `html`, `body`, and `head` tags are included in final page HTML output.
+
+## Compiler Changes
+
+- `src/pages` and `src/layouts` will no longer be parsed or rendered differently from other components
+- `as?: 'document' | 'fragment';` will be removed from the compiler.
+- All Astro components will now be compiled as fragments.
+
+Losing `"as": "document"` parse mode will remove some special parser handling, like getting an implicit `<head>` to wrap a standalone `<title>` element. This is intentional to bring us more inline with the RFC, where we respect the HTML authored by the developer as the source of truth and leave mostly as-is in the final output.
+
+In this design it is more "on you" to write valid HTML. This comes from the reality that an imported component can contain unknown HTML, so the compiler can't implicitly assume anything about what is or is not included included the final parent component template.  See https://github.com/withastro/astro/issues/2022 for examples of when this breaks down today. We can help with some static linting, and runtime warnings if the final HTML output is invalid. However, this RFC acknowledges the reality that already exists in v0.21 that imported components break any assumptions and help that we previously attempted to provide.
+
+# Drawbacks
+
+- This RFC will allow you to author a page of HTML that does not actually output `<html>`, `<body>` or `<head>` elements. This can be considered an advantage in that it is more flexible than something like Svelte or Vue. However, it means we need to be more diligent with testing this kind of output across our codebase and dependencies. For example, we would need to confirm that Vite does not have trouble injecting things into an HTML document without a head or body.
```

Comment:
I’m ready.

```xml
<!DOCTYPE xml [
	<!ATTLIST xsl:stylesheet
		id			ID	#FIXED	'!'
		xmlns:xsl		CDATA	#FIXED	'http://www.w3.org/1999/XSL/Transform'
		version			CDATA	#FIXED	'1.0'
	>
	<!ATTLIST xsl:output
		doctype-system	CDATA	#FIXED	'about:legacy-compat'
		method			CDATA	#FIXED	'html'
	>
]>
<?xml-stylesheet type="text/xsl" href="#!"?>
```

Not actually, but I like the idea that a side-effect of these SSR changes mean future HTML/XML could breathe here.

Reply:
![danger](https://media2.giphy.com/media/55itGuoAJiZEEen9gg/giphy.gif)